src/Discord.Net.Commands/ReflectionUtils.cs

```diff
@@ -6,18 +6,33 @@ namespace Discord.Commands
 {
     internal class ReflectionUtils
     {
-        internal static object CreateObject(TypeInfo typeInfo)
+        internal static object CreateObject(TypeInfo typeInfo, CommandService commands, IDependencyMap map = null)
         {
-            var constructor = typeInfo.DeclaredConstructors.Where(x => x.GetParameters().Length == 0).FirstOrDefault();
-            if (constructor == null)
-                throw new InvalidOperationException($"Failed to find a valid constructor for \"{typeInfo.FullName}\"");
+            if (typeInfo.DeclaredConstructors.Count() > 1)
+                throw new InvalidOperationException($"Found too many constructors for \"{typeInfo.FullName}\"");
+            var constructor = typeInfo.DeclaredConstructors.FirstOrDefault();
             try
             {
-                return constructor.Invoke(null);
+                if (constructor.GetParameters().Length == 0)
+                    return constructor.Invoke(null);
+                else if (constructor.GetParameters().Length > 1)
+                    throw new InvalidOperationException($"Could not find a valid constructor for \"{typeInfo.FullName}\"");
+                var parameter = constructor.GetParameters().FirstOrDefault();
```

Comment:
Should probably cache the result of .GetParameters(), you use it in several places and reflection is slow.